easybuild/easyblocks/generic/cmakemake.py

```diff
@@ -141,12 +141,18 @@ def build_type(self):
 
     def prepend_config_opts(self, config_opts):
         """Prepends configure options (-Dkey=value) to configopts ignoring those already set"""
+        # need to disable template resolution or it will remain the same for all runs
+        self.cfg.enable_templating = False
```

Comment:
We have a cleaner solution for this (since https://github.com/easybuilders/easybuild-framework/pull/3668):

```python
with self.cfg.disable_templating():
    ...

Reply:
My quick test shows that it is only the `cfg_configopts = self.cfg['configopts']` that needs to be done with templating disabled. So:
```py
        # need to disable template resolution or it will remain the same for all runs
        with self.cfg.disable_templating():
            cfg_configopts = self.cfg['configopts']
        # All options are of the form '-D<key>=<value>'
        new_opts = ' '.join('-D%s=%s' % (key, value) for key, value in config_opts.items()
                            if '-D%s=' % key not in cfg_configopts)
        self.cfg['configopts'] = ' '.join([new_opts, cfg_configopts])
```

Reply:
I agree here: Only the get needs to be guarded

Reply:
See https://github.com/easybuilders/easybuild-easyblocks/pull/2884

Reply:
done